### DISCUSSION TITLE
Nullable API

### DISCUSSION THREAD

**Discussion by inputfalken** [dotnet/runtime#54783] [Q&A] [answered]
Hi,

I wonder if there are any plans to have methods similar to the methods available for [Optional&lt;T&gt;](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) in Java. 
By using [nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references) and [nullable value types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types)?

By methods, I mainly think of these. 

* Filter
* FlatMap
* Map

**Comment by huoyaoyuan** [comment, marked as answer]
There's nearly no space for it in corelib. .NET mainly uses `bool TryXXX(out)` pattern.
If it would be added, the main benefit should be the `XXXOrDefault` series of methods. Your examples don't care about the type at all.

**Reply by inputfalken** [reply]
I see thanks for the response.
